docs/react-jss.md

```diff
@@ -587,4 +587,101 @@ const Button = () => {
 
 ## TypeScript
 
-TODO hooks support
+React-JSS enables you to supply types for your `ruleNames`, `props`/`data`, and `theme` both explicitly and implicitly.
+
+### Explicit Typing
+
+```tsx
+// Define Component
+type RuleNames = 'myButton' | 'myLabel'
+
+interface ButtonProps {
+  children?: React.ReactNode
+  spacing?: number
+  fontWeight?: string
+  labelColor?: string
+}
+
+interface CustomTheme {
+  background: string
+}
+
+const useStyles = createUseStyles<RuleNames, ButtonProps, CustomTheme>({
+  myButton: {
+    padding: ({spacing}) => spacing || 10
+  },
+  myLabel: ({theme, ...props}) => ({
+    display: 'block',
+    color: props.labelColor || 'red',
+    fontWeight: props.fontWeight || 'bold',
+    backgroundColor: theme.background || 'gray'
+  })
+})
+
+function Button({children, ...props}: ButtonProps): React.ReactElement {
+  const theme = useTheme<CustomTheme>()
+  const classes = useStyles({...props, theme})
+
+  return (
+    <button className={classes.myButton}>
+      <span className={classes.myLabel}>{children}</span>
+    </button>
+  )
+}
+
+// Create App
+const theme = {background: 'black'}
+
+function App(): React.ReactElement {
+  return (
+    <ThemeProvider theme={theme}>
+      <Button fontWeight="bold">Submit</Button>
+    </ThemeProvider>
+  )
+}
+```
+
+You only need to supply as much type information as you need. For instance, if you aren't using a `Theme`, then you don't need to supply a type for it. You'll notice that if you want to supply types for `Props` or `Theme`, you'll have to supply the type for `RuleNames` first. This means that if you want autocomplete for the `classes` that come from `useStyles`, you'll need to explicitly provide the strings you'll be using. This restriction is due to a limitation in TypeScript [that will hopefully get resolved soon](https://github.com/microsoft/TypeScript/pull/26349).
```

Comment:
I would consider extending the last sentence to say _"or, see the next section for a workaround"_. 

Just a thought, in case some people stop reading at the end of that section. There seems to be quite a misunderstanding about this point (https://github.com/cssinjs/jss/pull/1460#issuecomment-799478204, #1469)

Reply:
Yeah that's a good point. I didn't expect this to be such a large source of confusion. I tried to add some additional words like you said for clarity.

My recent commit has changes that address both of your suggestions. If you have free time, let me know what you think.

Reply:
Looks great :+1: